src/App.tsx

```diff
@@ -1,10 +1,51 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Errors } from './types/Errors';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './components/ErrorMessage';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+
+const getFileteredTodos = (todos: Todo[], filter: Filter): Todo[] => {
+  return todos.filter(todo => {
+    switch (filter) {
+      case Filter.Completed:
+        return todo.completed;
+      case Filter.Active:
+        return !todo.completed;
+      default:
+        return true;
+    }
+  });
+};
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors>(Errors.Default);
+  const [filterOption, setFilterOption] = useState<Filter>(Filter.All);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodosFromServer)
+      .catch(() => setErrorMessage(Errors.Loading))
+      .finally(() =>
+        setTimeout(() => {
+          setErrorMessage(Errors.Default);
+        }, 3000),
```

Comment:
- You should remove the error only in case you have it
- And as you have to do it for all errors you can create a handler and reuse it
```typescript
const handleError = (errorMessage: Errors) => {
  setError(errorMessage);
  setTimeout(() => setErrorMessage(errorMessage), 3000);
};
```